packages/hash/api/src/model/knowledge/page.model.ts

```diff
@@ -0,0 +1,481 @@
+  LinkModel,
+  UserModel,
+  OrgModel,
+} from "..";
+import { WORKSPACE_TYPES } from "../../graph/workspace-types";
+
+type PageModelCreateParams = Omit<
+  EntityModelCreateParams,
+  "properties" | "entityTypeModel"
+> & {
+  title: string;
+  summary?: string;
+  prevIndex?: string;
+  initialBlocks?: BlockModel[];
+};
+
+/**
+ * @class {@link PageModel}
+ */
+export default class extends EntityModel {
+  static fromEntityModel(entity: EntityModel): PageModel {
+    if (
+      entity.entityTypeModel.schema.$id !==
+      WORKSPACE_TYPES.entityType.page.schema.$id
+    ) {
+      throw new Error(
+        `Entity with id ${entity.entityId} is not a workspace page`,
+      );
+    }
+
+    return new PageModel(entity);
+  }
+
+  /**
+   * Get a workspace page entity by its entity id.
+   *
+   * @param params.entityId - the entity id of the page
+   */
+  static async getPageById(
+    graphApi: GraphApi,
+    params: { entityId: string },
+  ): Promise<PageModel> {
+    const entity = await EntityModel.getLatest(graphApi, params);
+
+    return PageModel.fromEntityModel(entity);
+  }
+
+  /**
+   * Create a workspace page entity.
+   *
+   * @param params.title - the title of the page
+   */
+  static async createPage(
+    graphApi: GraphApi,
+    params: PageModelCreateParams,
+  ): Promise<PageModel> {
+    const { title, summary, prevIndex, accountId } = params;
+
+    const index = generateKeyBetween(prevIndex ?? null, null);
+
+    const properties: object = {
+      [WORKSPACE_TYPES.propertyType.title.baseUri]: title,
+      [WORKSPACE_TYPES.propertyType.summary.baseUri]: summary,
+      [WORKSPACE_TYPES.propertyType.index.baseUri]: index,
+    };
+
+    const entityTypeModel = WORKSPACE_TYPES.entityType.page;
+
+    const entity = await EntityModel.create(graphApi, {
+      accountId,
+      properties,
+      entityTypeModel,
+    });
+
+    const page = PageModel.fromEntityModel(entity);
+
+    const initialBlocks =
+      params.initialBlocks && params.initialBlocks.length > 0
+        ? params.initialBlocks
+        : [
+            await BlockModel.createBlock(graphApi, {
+              accountId,
+              componentId: "https://blockprotocol.org/blocks/@hash/paragraph",
+              blockData: await EntityModel.create(graphApi, {
+                accountId,
+                properties: {
+                  [WORKSPACE_TYPES.propertyType.tokens.baseUri]: [],
+                },
+                entityTypeModel: WORKSPACE_TYPES.entityType.text,
+              }),
+            }),
+          ];
+
+    for (const block of initialBlocks) {
+      await page.insertBlock(graphApi, {
+        block,
+        insertedBy: accountId,
+      });
+    }
+
+    return page;
+  }
+
+  /**
+   * Get all the pages in an account.
+   *
+   * @param params.account - the user or org whose pages will be returned
+   */
+  static async getAllPagesInAccount(
+    graphApi: GraphApi,
+    params: {
+      account: UserModel | OrgModel;
```

Comment:
I think we wanted to be specific to add `Model` when dealing with the model classes
```suggestion
      accountModel: UserModel | OrgModel;
```